### PR TITLE
Furnace Test Example: added transmission to the test

### PR DESCRIPTION
**Pull request by WestLangley**
... plus other minor fixes.

### CODE REVIEW

**Comment by WestLangley** [comment]
It appears that orthographic camera, PMREM, and/or transmission are mutually incompatible, so I reverted the camera back to a perspective one.

**Comment by mrdoob** [comment]
Thanks!

**Comment by WestLangley** [comment]
@Mugen87 If `transmission` is set to 1, then this example renders incorrectly on the first rendering. Move the mouse to force another render.

I expect all the promises in this example are not required, but even after removing them and simplifying the example to follow the typical three.js coding pattern, the issue remains. It seems like transmission thinks the background is black...

File: examples/webgl_furnace_test.html [hunk 1]
```diff
@@ -48,32 +48,33 @@
...
 
 				// scene
 
 				scene = new THREE.Scene();
 
 				// camera
-
-				camera = new THREE.OrthographicCamera( frustumSize * aspect / - 2, frustumSize * aspect / 2, frustumSize / 2, frustumSize / - 2, 1, 30 );
-				camera.position.set( 0, 0, 10 );
+				camera = new THREE.PerspectiveCamera( 40, aspect, 1, 30 );
```
Comment (Mugen87) [inline examples/webgl_furnace_test.html:76]: The code in `onWindowResize()` still thinks an orthographic camera is used.

**Comment by Mugen87** [comment]
>  If transmission is set to 1, then this example renders incorrectly on the first rendering. Move the mouse to force another render.

I can reproduce. However it's not clear to me whether this is only an issue in the first frame or if the transmission implementation is frame-late in general.

@takahirox Do you mind having a look? Here is a test case: https://jsfiddle.net/psjwzyft/

**Comment by mrdoob** [comment]
Doing this shows that it's a frame late indeed:

```js
function render() {

  renderer.render(scene, camera);
  renderer.render(scene, camera);

}
```

File: examples/webgl_furnace_test.html (hunk 1 above)
Comment (WestLangley) [reply examples/webgl_furnace_test.html:76]: fixed!

**Comment by takahirox** [comment]
> @takahirox Do you mind having a look? Here is a test case: https://jsfiddle.net/psjwzyft/

OK, let me try hopefully this weekend

**Comment by takahirox** [comment]
I investigated the problem this weekend but I couldn't find the root issue. But I found some weird behaviors. Does anyone here have any clues from them?

* In `webgl_furnace_test` example with `transmission=1`, the problem was disappeared if I set `renderer.outputEncoding = THREE.sRGBEncoding`
* In `webgl_furnace_test` example with `transmission=1`, the problem was disappeared if I set `scene.background = new THREE.Color( COLOR );` instead of `hdrEquirect`
* The problem can reproducible even in `webgl_materials_physical_transmission` example if I comment out the `hdrEquirect.mapping = THREE.EquirectangularReflectionMapping;` and `renderer.outputEncoding = THREE.sRGBEncoding;` lines

So, sounds like the problem happens if `outputEncoding` is not `linear (default)` and `scene.background` is not cube map?

**Comment by Mugen87** [comment]
After merging #22394, the following warning appears now in the first render call of the fiddle:

> [.WebGL-0x7ffe34856400]RENDER WARNING: there is no texture bound to the unit 0

**Comment by Mugen87** [comment]
Is seems the issue can be fixed by forcing a uniform update of the scene's background before the opaque objects are rendered for the transmissive pass. Meaning before the following line:

https://github.com/mrdoob/three.js/blob/57b7a9b5aa4ca4339419bc10bd531fcf1f0cc546/src/renderers/WebGLRenderer.js#L1269

I add this hack.
```js
opaqueObjects[ 0 ].material.uniformsNeedUpdate = true; // the first render item is the skybox
```

**Pull request merged**